packages/angular_devkit/architect/src/api.ts

```diff
@@ -0,0 +1,230 @@
+   */
+  stop(): Promise<void>;
+}
+
+/**
+ * The context received as a second argument in your builder.
+ */
+export interface BuilderContext {
+  /**
+   * Unique amongst contexts. Contexts instances are not guaranteed to be the same (but it could
+   * be the same context), and all the fields in a context could be the same, yet the builder's
+   * context could be different. This is the same ID as the corresponding run.
+   */
+  id: number;
+
+  /**
+   * The builder info that called your function. Since the builder info is from the builder.json
+   * (or the host), it could contain information that is different than expected.
+   */
+  builder: BuilderInfo;
+
+  /**
+   * A logger that appends messages to a log. This could be a separate interface or completely
+   * ignored. `console.log` could also be completely ignored.
+   */
+  logger: logging.LoggerApi;
+
+  /**
+   * The absolute workspace root of this run. This is a system path and will not be normalized;
+   * ie. on Windows it will starts with `C:\\` (or whatever drive).
+   */
+  workspaceRoot: string;
+
+  /**
+   * The current directory the user is in. This could be outside the workspace root. This is a
+   * system path and will not be normalized; ie. on Windows it will starts with `C:\\` (or
+   * whatever drive).
+   */
+  currentDirectory: string;
+
+  /**
+   * The target that was used to run this builder.
+   * Target is optional if a builder was ran using `scheduleBuilder()`.
+   */
+  target?: Target;
+
+  /**
+   * Schedule a target in the same workspace. This can be the same target that is being executed
+   * right now, but targets of the same name are serialized.
+   * Running the same target and waiting for it to end will result in a deadlocking scenario.
+   * Targets are considered the same if the project, the target AND the configuration are the same.
+   * @param target The target to schedule.
+   * @param overrides A set of options to override the workspace set of options.
+   * @return A promise of a run. It will resolve when all the members of the run are available.
+   */
+  scheduleTarget(
+    target: Target,
+    overrides?: json.JsonObject,
+  ): Promise<BuilderRun>;
+
+  /**
+   * Schedule a builder by its name. This can be the same builder that is being executed.
+   * @param builderName The name of the builder, ie. its `packageName:builderName` tuple.
+   * @param options All options to use for the builder (by default empty object). There is no
+   *     additional options added, e.g. from the workspace.
+   * @return A promise of a run. It will resolve when all the members of the run are available.
+   */
+  scheduleBuilder(
+    builderName: string,
+    options?: json.JsonObject,
+  ): Promise<BuilderRun>;
+
+  /**
+   * Set the builder to running. This should be used if an external event triggered a re-run,
+   * e.g. a file watched was changed.
+   */
+  running(): void;
```

Comment:
This should have a verb.  Currently from the name I'd expect it to return a boolean signifying whether it is running.

`setRunning`? `reportRunning`?  -- depending on what it actually does.

Same for status and progress.  I believe this would help to make the API more intuitive.
There is already `schedule` as a verb/action above.  I think having `report` as well would nicely define the set of actions that can be performed via the context.  A builder can either schedule additional executions or report on its behavior.

Reply:
Yes, I agree, and I scratched my head a bit on that one. `report` is good.